Connect backend to frontend, polish, & finalize project

Skywalker31415 opened the pull request:
Needs a lot of display modifications but the basic data is there- it displays weather and updates based on city. Does not currently change the image based on weather because I need to figure out a way to parse the complex strings. Also does not check to make sure city is valid.

Commit:
Backend Data Retrieval

Commit:
Backend Data Retrieval

Commit:
Added in backend data parsing algorithms, and created a route for communication between the frontend and backend.

Commit:
Used jsonify from flask to display weather data for frontend integration.

Commit:
Added in separate test case file, finalizing that the backend codebase works as expected, and will work on connecting Python to JavaScript.

Commit:
Backend codebase completed

Commit:
Backend codebase completed

Commit:
Backend codebase completed

Commit:
Backend codebase completed

Timothy-Gonzalez committed:
Remove __pycache__

Timothy-Gonzalez committed:
Add __pycache__ to .gitignore

Skywalker31415 committed:
api pulling

Commit:
Rewrote parsing algorithm and fixed inputs for Flask route

Commit:
Rewrote parsing algorithm and fixed inputs for Flask route

Commit:
Merge branch 'main' of https://github.com/CS222-UIUC/the-weather-app into eeshan

Commit:
Merge branch 'eeshan' of https://github.com/CS222-UIUC/the-weather-app into eeshan

Commit:
Rewrote the entire backend

Skywalker31415 committed:
committing the basic stuff I added before merge

Skywalker31415 committed:
committing the basic stuff I added before merge

Skywalker31415 committed:
Merge branch 'main' into connectapi

Skywalker31415 committed:
Merge branch 'eeshan' into connectapi

Skywalker31415 committed:
connecting api

Skywalker31415 committed:
Connects backend API to frontend display

Timothy-Gonzalez commented on backend/app.py:
```diff
@@ -11,6 +11,12 @@
+    geolocator = Nominatim(user_agent="app", timeout=10)
+    location = geolocator.geocode(location)
+    locations = [location.latitude, location.longitude]
+    return locations
```
This looks like a bad merge - this shouldn't need to be added

Timothy-Gonzalez replied:
Resolved by 32c5864.

Timothy-Gonzalez commented on backend/server.py:
```diff
@@ -0,0 +1,11 @@
+    return {"members": "member1"}
+
+if __name__ == "__main__":
+    app.run(debug = True)
```
What is this?

Timothy-Gonzalez replied:
RE: Apparently was testing code, can be safely removed

Timothy-Gonzalez replied:
Resolved by a37d32d.

Timothy-Gonzalez commented on frontend/src/component/WeatherForecastCurrent.tsx:
```diff
@@ -1,19 +1,53 @@
+//, high: string, low: string, weather: string
+type MyProps = {city: string, unit: string, data:string[]};
+type MyState = { current_temp: number };
+export default class WFC extends React.Component<MyProps,MyState>{    
```
We need to figure out how to convert forecast into one of these types. Also, this should probably be an enum, with a pairing to the image locations.

Timothy-Gonzalez replied:
Resolved by a672642.

Timothy-Gonzalez commented on package-lock.json:
```diff
@@ -0,0 +1,6 @@
+  "lockfileVersion": 3,
+  "requires": true,
+  "packages": {}
+}
```
This shouldn't exist, see `frontend/package-lock.json`.

Timothy-Gonzalez replied:
Resolved by d9f892d.

Timothy-Gonzalez commented on backend/package-lock.json:
```diff
@@ -0,0 +1,6 @@
+  "lockfileVersion": 3,
+  "requires": true,
+  "packages": {}
+}
```
This shouldn't exist, see `frontend/package-lock.json`.

Timothy-Gonzalez replied:
Resolved by d9f892d.

Timothy-Gonzalez reviewed (changes requested):
Overall, great job! Does what it needs to and handles the data pretty well. It's great that you got location changing working too!

This just needs a bit of clean up, and then everything should be great!

Other things to do:
- [x] Rename classes and refractor out
- [x] Add styling

Timothy-Gonzalez committed:
Remove bad merge artifact

Timothy-Gonzalez committed:
Remove testing file

Timothy-Gonzalez committed:
Remove extraneous package locks

Timothy-Gonzalez committed:
Remove unused tests

Timothy-Gonzalez committed:
Run formatter

Timothy-Gonzalez committed:
Remove graph

Timothy-Gonzalez committed:
Refractor out report type, add better styling

Timothy-Gonzalez committed:
Improve look of hourly forecast

Timothy-Gonzalez committed:
Shift warnings list to be aligned

Timothy-Gonzalez committed:
Fix formatting of warnings

Timothy-Gonzalez committed:
Convert °F -> °, & add disclaimer

Timothy-Gonzalez committed:
Factor out weather type & add image identification

Timothy-Gonzalez committed:
Improve weekly forecast styling

Timothy-Gonzalez committed:
Center forecasts

Timothy-Gonzalez committed:
Add coordinate support to backend

Timothy-Gonzalez committed:
Add local location option & default

Timothy-Gonzalez commented:
Alright, this should be basically it. Test this locally and make sure it works for you. Once you **both** have approved this, then merge it.

Timothy-Gonzalez committed:
Fix no response error from no observation

If the observed value was null, don't try to convert with pytemp as that
will cause an infinite yield

Timothy-Gonzalez committed:
Fix max temperature

Timothy-Gonzalez committed:
Try all stations that exist by distance

Timothy-Gonzalez committed:
Fix status checking

Timothy-Gonzalez committed:
Revert "Fix status checking"

This reverts commit d4d0250651c6c16a84d37bd253bb4b481dd761c8.

Turns out I was right the first time

Timothy-Gonzalez committed:
Filter out bad observations when finding station

Skywalker31415 committed:
Added climate change data and clothing advice

Skywalker31415 committed:
Adjusted position of graph

Timothy-Gonzalez commented on frontend/src/data/WeatherType.ts:
```diff
@@ -55,3 +67,27 @@ export function GetImageSrcFor(forecast: string): string {
+        const weatherTypeData = WeatherTypeData[weatherType]
+        for (const word of weatherTypeData.words) {
+            if (forecast.toLowerCase().includes(word)) {
+                return weatherTypeData.cc_src
```
There should be a GetWeatherTypeForForecast method to reduce duplication here

Timothy-Gonzalez commented:
Also, I'll look at this more later and actually go through it, don't merge until then.

Timothy-Gonzalez committed:
Add formatting

Timothy-Gonzalez committed:
Refractor WeatherType

Timothy-Gonzalez committed:
Remove old print

Timothy-Gonzalez committed:
Fix padding & scroll

Timothy-Gonzalez committed:
Expose loading & error statuses to user

Timothy-Gonzalez reviewed (approved):
It's ready.
